services/web/server/src/simcore_service_webserver/garbage_collector/_core_orphans.py

```diff
@@ -38,17 +38,21 @@ async def _remove_service(
         save_service_state = False
     else:
         try:
-            if await get_user_role(app, user_id=service.user_id) <= UserRole.GUEST:
-                save_service_state = False
-            else:
-                save_service_state = await has_user_project_access_rights(
+            user_role: UserRole = await users_service.get_user_role(
+                app, user_id=service.user_id
+            )
+        except (UserNotFoundError, ValueError):
+            save_service_state = False
+        else:
+            save_service_state = (
+                user_role > UserRole.GUEST
+                and await has_user_project_access_rights(
                     app,
                     project_id=service.project_id,
                     user_id=service.user_id,
                     permission="write",
                 )
-        except (UserNotFoundError, ValueError):
-            save_service_state = False
```

Comment:
Why was this changed? I find it wrong.

If there is no user you can't be found, saving the state will create an error. This will break dynamic services, please revert the change.
One edge case would be when an anonymous user is deleted.
Regarding the `ValueError`,I have not idea but I would not remove it, since it will/might cause unexpected behavours hard to debug/predict.